pkg/compose/build_buildkit.go

```diff
@@ -34,8 +35,11 @@ import (
 	"github.com/moby/buildkit/client"
 )
 
-func (s *composeService) doBuildBuildkit(ctx context.Context, service string, opts build.Options, p *buildx.Printer) (string, error) {
-	b, err := builder.New(s.dockerCli)
+func (s *composeService) doBuildBuildkit(ctx context.Context, service string, opts build.Options, p *buildx.Printer, builderName string) (string, error) {
+	if builderName == "" {
+		builderName = os.Getenv("BUILDX_BUILDER")
+	}
```

Comment:
Not critical, but I think it might be a bit cleaner/separation of concerns to do the fallback env lookup in the `cmd/` package parts and use the value as-is here

Reply:
Good catch, thanks Milas